### PR TITLE
Stop VATZ process if all notification channels fail to send notifications

### LINKED ISSUE #500: when dispatcher config input VATZ must not have panic when config input isn't appropriate
### Checklist
- [x] Bug from Service (Vatz)
- [ ] Bug from Plugin (Vatz) 
- [ ] Malfunction on CI/CD 
- [ ] etc(e.g, Hardware version doesn't match, )

---

1. Make build
2. leave a blank for dispatchers default.yaml 
3. start VATZ
4. Error occurs

```
 ./vatz start                   
2023-09-29T23:16:08-05:00 INF Initialize Server module=main
2023-09-29T23:16:08-05:00 INF Start VATZ Server on Listening Port: :9090 module=main
2023-09-29T23:16:08-05:00 INF start metric server: 127.0.0.1:18080 module=main
2023-09-29T23:16:08-05:00 INF start rpc server module=rpc
2023-09-29T23:16:08-05:00 INF start gRPC gateway server 127.0.0.1:19091 module=rpc
2023-09-29T23:16:08-05:00 INF start gRPC server 127.0.0.1:19090 module=rpc
2023-09-29T23:16:12-05:00 ERR Channel(Discord): Connection failed due to Invalid discord webhook address module=dispatcher
2023-09-29T23:16:13-05:00 ERR Channel(Pagerduty): Connection failed due to Error: HTTP response with status code: 400, JSON unmarshal object body failed: invalid character 'I' looking for beginning of value, body: Invalid routing key module=dispatcher
panic: interface conversion: interface {} is nil, not bool

goroutine 23 [running]:
github.com/dsrvlabs/vatz/manager/dispatcher.(*telegram).SendNotification(0xc000276380, {{0x1009641a4, 0xa}, 0x4, {0x100967dc7, 0x10}, 0x3, {0xc000256e70, 0xb}, 0xc00038e3f0})
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/manager/dispatcher/telegram.go:121 +0x91f
github.com/dsrvlabs/vatz/manager/healthcheck.(*healthChecker).PluginHealthCheck(0x100f45b80, {0x100a75ba8, 0xc000136008}, {0x100a72458, 0xc00033d1a0}, {{0xc000256e70, 0xb}, {0xc000256e57, 0x9}, 0x4, ...}, ...)
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/manager/healthcheck/general_healthcheck.go:69 +0x5b8
github.com/dsrvlabs/vatz/cmd.multiPluginExecutor({{0xc000256e70, 0xb}, {0xc000256e57, 0x9}, 0x4, 0x5, 0x2329, {0xc00033c6f0, 0x1, 0x1}}, ...)
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/cmd/start.go:146 +0x4d9
created by github.com/dsrvlabs/vatz/cmd.startExecutor
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/cmd/start.go:125 +0x87
 ✘ dongyookang DK 👑   ~/ffplay/GolandProjects/xellos/vatz   ISSUE-312-re ±
 

```

Throw error is okay, but not panic. 

image
image

### LINKED ISSUE #519: Telegram bugs that throw error when it send notification
### Checklist
- [x] Bug from Service (Vatz)
- [ ] Bug from Plugin (Vatz) 
- [ ] Malfunction on CI/CD 
- [ ] etc(e.g, Hardware version doesn't match, )

---

1. Start VATz with Telegram

```
 ./vatz start                       
2023-10-20T23:43:56-05:00 INF Initialize Server module=main
2023-10-20T23:43:56-05:00 INF Start VATZ Server on Listening Port: :9090 module=main
2023-10-20T23:43:56-05:00 INF Client successfully connected to localhost:9001 (plugin:cpu_monitor). module=util
2023-10-20T23:43:56-05:00 INF start metric server: 127.0.0.1:18080 module=main
2023-10-20T23:43:56-05:00 INF start rpc server module=rpc
2023-10-20T23:43:56-05:00 INF start gRPC gateway server 127.0.0.1:19091 module=rpc
2023-10-20T23:43:56-05:00 INF start gRPC server 127.0.0.1:19090 module=rpc
2023-10-20T23:43:57-05:00 INF Client successfully connected to localhost:9001 (plugin:cpu_monitor). module=util
2023-10-20T23:44:26-05:00 INF Executor send request to cpu_monitor module=executor
2023-10-20T23:44:26-05:00 INF response: SUCCESS module=executor
panic: interface conversion: interface {} is nil, not bool

goroutine 39 [running]:
github.com/dsrvlabs/vatz/manager/dispatcher.(*telegram).SendNotification(0xc000116580, {{0x1009632b1, 0xa}, 0x4, {0x100966ed4, 0x10}, 0x3, {0xc000042560, 0xb}, 0xc00011ba70})
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/manager/dispatcher/telegram.go:121 +0x91f
github.com/dsrvlabs/vatz/manager/healthcheck.(*healthChecker).PluginHealthCheck(0x100f46060, {0x100a74f48, 0xc0001ae008}, {0x100a717f8, 0xc00007b6d0}, {{0xc000042560, 0xb}, {0xc000042547, 0x9}, 0xf, ...}, ...)
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/manager/healthcheck/general_healthcheck.go:69 +0x5b8
github.com/dsrvlabs/vatz/cmd.multiPluginExecutor({{0xc000042560, 0xb}, {0xc000042547, 0x9}, 0xf, 0x1e, 0x2329, {0xc00007ac30, 0x1, 0x1}}, ...)
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/cmd/start.go:145 +0x4d9
created by github.com/dsrvlabs/vatz/cmd.startExecutor
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/cmd/start.go:124 +0xc7

```

fix bugs while sending notification to telegram.

### PR DESCRIPTION
**Pull request by xellos00**
### 1. Type of change

Please delete options that are not relevant.

- [ ] New feature
- [x] Enhancement
- [ ] Bug/fix (non-breaking change which fixes an issue)
- [ ] others (anything other than above)

---

**Related:** # (issue)
close #500 
close #519

**Summary**
must stop the process if all notification channels are fail to send notifications. 

---

### CODE REVIEW

**Comment by xellos00** [comment]
```
 ✘ dongyookang DK 🍻   ~/ffplay/GolandProjects/xellos/vatz   ISSUE-500
 make coverage
?       github.com/dsrvlabs/vatz        [no test files]
?       github.com/dsrvlabs/vatz/manager/types  [no test files]
?       github.com/dsrvlabs/vatz/mocks  [no test files]
ok      github.com/dsrvlabs/vatz/cmd    2.104s  coverage: 23.0% of statements
ok      github.com/dsrvlabs/vatz/manager/api    0.643s  coverage: 0.0% of statements [no tests to run]
ok      github.com/dsrvlabs/vatz/manager/config 0.493s  coverage: 77.8% of statements
ok      github.com/dsrvlabs/vatz/manager/dispatcher     0.861s  coverage: 6.9% of statements
ok      github.com/dsrvlabs/vatz/manager/executor       1.171s  coverage: 67.9% of statements
ok      github.com/dsrvlabs/vatz/manager/healthcheck    1.396s  coverage: 50.0% of statements
ok      github.com/dsrvlabs/vatz/manager/plugin 7.230s  coverage: 51.0% of statements
ok      github.com/dsrvlabs/vatz/monitoring/prometheus  1.861s  coverage: 0.0% of statements
ok      github.com/dsrvlabs/vatz/rpc    3.276s  coverage: 67.2% of statements
ok      github.com/dsrvlabs/vatz/utils  2.098s  coverage: 3.6% of statements
 dongyookang DK 🍻   ~/ffplay/GolandProjects/xellos/vatz   ISSUE-500

```

**Review by meetrick — APPROVED** [review]
LGTM

**Review by meetrick — APPROVED** [review]
LGTM

File: manager/dispatcher/telegram.go [hunk 1]
```diff
@@ -112,6 +113,13 @@ Plugin Name: <em>%s</em>
 		return err
 	}
 	defer response.Body.Close()
+
+	if response.StatusCode < 200 || response.StatusCode > 299 {
+		log.Error().Str("module", "dispatcher").Msgf("Channel(Telegram): Error in Response with Error code: %d", response.StatusCode)
+		errorMessage := fmt.Sprintf("REST API Error with HTTP response status code: %d", response.StatusCode)
```
Comment (rootwarp) [inline manager/dispatcher/telegram.go:119]: Suggest to use `fmt.Errorf` function to create error instance.

Comment (xellos00) [reply manager/dispatcher/telegram.go]: I've updated as you mentioned. Can you review again?

Comment (rootwarp) [reply manager/dispatcher/telegram.go]: awesome thanks for considering.

**Review by rootwarp — APPROVED** [review]
LGTM

**Pull request merged**